### PR TITLE
pooling versions of get-schema and get-cred-def cmds

### PR DESCRIPTION
**Pull request by techlab-lainio**

### CODE REVIEW

File: cmd/agent/getcreddef.go [hunk 1]
```diff
@@ -39,11 +42,25 @@ var getCredDefCmd = &cobra.Command{
 		defer cancel()
 
 		agent := agency.NewAgentServiceClient(conn)
-		r, err := agent.GetCredDef(ctx, &agency.CredDef{
-			ID: CredDefID,
-		})
-		err2.Check(err)
-		fmt.Println(r.Data) // plain output for pipe/filter style
+
+		for left := poolTimeout - wait; left >= 0; left -= wait {
```
Comment (lauravuo-techlab) [inline cmd/agent/getcreddef.go:46]: pollTimeout?

File: cmd/agent/getcreddef.go [hunk 2]
```diff
@@ -57,7 +74,19 @@ func init() {
 	defer err2.Catch(func(err error) {
 		fmt.Println(err)
 	})
-	getCredDefCmd.Flags().StringVarP(&CredDefID, "id", "i", "", "credDef ID")
+
+	flags := getCredDefCmd.Flags()
+	flags.StringVarP(&CredDefID, "id", "i", "",
+		cmd.FlagInfo("credDef ID", "", credDefEnvs["id"]))
+
+	flags.DurationVarP(&wait, "wait", "w", time.Second, "sleep between polls, 0 == no pool")
```
Comment (lauravuo-techlab) [inline cmd/agent/getcreddef.go:82]: no poll?

File: cmd/agent/getcreddef.go [hunk 3]
```diff
@@ -57,7 +74,19 @@ func init() {
...
+	flags.DurationVar(&poolTimeout, "timeout", 10*time.Second, "how long to poll until give up")
```
Comment (lauravuo-techlab) [inline cmd/agent/getcreddef.go:83]: pollTimeout?

File: cmd/agent/getschema.go [hunk 4]
```diff
@@ -39,26 +42,53 @@ var getSchemaCmd = &cobra.Command{
...
 
 		agent := agency.NewAgentServiceClient(conn)
-		r, err := agent.GetSchema(ctx, &agency.Schema{
-			ID: schemaID,
-		})
-		err2.Check(err)
-		// plain output for script-friendlyness of the command
-		fmt.Println(r.Data)
+
+		for left := poolTimeout - wait; left >= 0; left -= wait {
```
Comment (lauravuo-techlab) [inline cmd/agent/getschema.go:46]: pollTimeout?

File: cmd/agent/getschema.go [hunk 5]
```diff
@@ -39,26 +42,53 @@ var getSchemaCmd = &cobra.Command{
...
 
 		return nil
 	},
 }
 
 var (
 	schemaID string
+
+	wait        time.Duration
+	poolTimeout time.Duration
```
Comment (lauravuo-techlab) [inline cmd/agent/getschema.go:73]: pollTimeout?

File: cmd/agent/getschema.go [hunk 6]
```diff
@@ -39,26 +42,53 @@ var getSchemaCmd = &cobra.Command{
...
 	defer err2.Catch(func(err error) {
 		fmt.Println(err)
 	})
-	getSchemaCmd.Flags().StringVarP(&schemaID, "schema-id", "i", "", "schema ID")
+
+	flags := getSchemaCmd.Flags()
+	flags.StringVarP(&schemaID, "schema-id", "i", "",
+		cmd.FlagInfo("schema ID", "", getSchemaEnvs["schema-id"]))
+
+	flags.DurationVarP(&wait, "wait", "w", time.Second, "sleep between polls, 0 == no pool")
```
Comment (lauravuo-techlab) [inline cmd/agent/getschema.go:85]: no poll?

**Review by lauravuo-techlab — APPROVED** [review]
Originally thought that the reading and polling would be added to the create command, so that client could be sure that the item is created and readable. This goes as well, but there will be a need to separately read the item after creation.

**Comment by techlab-lainio** [comment]
> Originally thought that the reading and polling would be added to the create command, so that client could be sure that the item is created and readable. This goes as well, but there will be a need to separately read the item after creation.

By doing this with **UNIX pipe-and-filter**, simplifies error handling a lot, and allows reuse. For example, you just want to create cred def but you mistyped your schemaID you can wait far too long to get the feedback. Naturally, when you are just reading a schema or cred def you know that it can poll if you don't prevent it with the flags. An of course, when it's atomic at this level you can reuse where ever you want in your scripts.

From client side it's ok that `create` functions validate inputs, but waiting for them is not the correct place especially when it's not coming from pipe but from ledger.

**Pull request merged**